core/src/main/java/org/elasticsearch/index/query/HasChildQueryBuilder.java

```diff
@@ -76,40 +72,23 @@
     private static final ParseField INNER_HITS_FIELD = new ParseField("inner_hits");
 
     private final QueryBuilder<?> query;
-
     private final String type;
+    private final ScoreMode scoreMode;
+    private final InnerHitBuilder innerHitBuilder;
+    private int minChildren;
+    private int maxChildren;
```

Comment:
QueryBuilders should always set the defaults since they can be constructed without ever passing through the parser (Java API)